numpy/lib/npyio.py

```diff
@@ -914,68 +914,10 @@ def loadtxt(fname, dtype=float, comments='#', delimiter=None,
            [ 19.22,  64.31],
            [-17.57,  63.94]])
     """
-    # Type conversions for Py3 convenience
-    if comments is not None:
-        if isinstance(comments, (str, bytes)):
-            comments = [comments]
-        comments = [_decode_line(x) for x in comments]
-        # Compile regex for comments beforehand
-        comments = (re.escape(comment) for comment in comments)
-        regex_comments = re.compile('|'.join(comments))
-
-    if delimiter is not None:
-        delimiter = _decode_line(delimiter)
```

Comment:
Could you perhaps do the moving of all these lines as a separate PR? It makes it really quite hard to review the bugfix with it merged into this one.

Reply:
OK, I'll create a maintenance PR for rearranging the code, and then follow up with the bug fix.

Reply:
Ideal, thanks!

Reply:
The maintenance PR is https://github.com/numpy/numpy/pull/16633